packages/nx/src/plugins/js/utils/register.ts

```diff
@@ -99,7 +99,14 @@ export function getTranspiler(compilerOptions: CompilerOptions) {
   }
 
   compilerOptions.lib = ['es2021'];
-  compilerOptions.module = ts.ModuleKind.CommonJS;
+  // Don't override module if it's set to node16 or nodenext
+  // Otherwise custom workspace eslint rules will fail with typescript 5.2
+  const preserveModuleOption =
+    compilerOptions.module === ts.ModuleKind.Node16 ||
+    compilerOptions.module === ts.ModuleKind.NodeNext;
+  compilerOptions.module = preserveModuleOption
+    ? compilerOptions.module
+    : ts.ModuleKind.CommonJS;
```

Comment:
Why limit to just the latest Node?
I would suggest accepting any value the user might be passing.

```suggestion
  compilerOptions.module = compilerOptions.module ||
    ts.ModuleKind.CommonJS;
```

Reply:
If the user sets the module value to be es2015 or es2020 though, the output will be in native ESM which requires running ts-node with the `--esm` option (and probably other updates to import paths etc)

Reply:
Sure, but we shouldn't be putting any constraints on users. 

That is what the enforced CommonJS that you are fixing with this PR was doing.

Reply:
OK I think it will be fine, I checked and all nx generators explicitly set `"module": "commonjs"` for workspace plugins tsconfig's etc, so it should only be breaking for users that overwrote the value